### ISSUE TITLE
Better code colouring with literal strings

### ISSUE THREAD

**Issue by ajabadi** [rstudio/rstudio#6788]
It would be great if RStudio could correctly recognise the code with the literal strings which include quote/unquotes:

```r
r"("""" This is a valid literal string up until paranthesis-unquote """"")"

a <- 4

options()$stringsAsFactors
```
Which displays:

![image](https://user-images.githubusercontent.com/28503057/80790069-34e3ee00-8bd1-11ea-957f-d833fb8f6f5b.png)

Although it does not have any issue running it.

I'm using  RStudio 1.3.911

``` r
sessionInfo()
#> R version 4.0.0 beta (2020-04-14 r78227)
#> Platform: x86_64-apple-darwin17.0 (64-bit)
#> Running under: macOS Catalina 10.15
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_4.0.0  magrittr_1.5    tools_4.0.0     htmltools_0.4.0
#>  [5] yaml_2.2.1      Rcpp_1.0.4.6    stringi_1.4.6   rmarkdown_2.1  
#>  [9] highr_0.8       knitr_1.28      stringr_1.4.0   xfun_0.13      
#> [13] digest_0.6.25   rlang_0.4.5     evaluate_0.14
```

<sup>Created on 2020-05-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by kevinushey** [comment]
Implementation note: this would likely be handled with some new tokenization rules here.

https://github.com/rstudio/rstudio/blob/04816401ff8af92c70a385d41b68ebe05cd76d7f/src/gwt/acesupport/acemode/r_highlight_rules.js#L241-L264

We'd need to use a stateful rule, as we'd need to capture the way the raw string was created to infer what would be used to close it.

Here's an example where we have a stateful rule for strings with YAML (where we need to track the indent):

https://github.com/rstudio/rstudio/blob/04816401ff8af92c70a385d41b68ebe05cd76d7f/src/gwt/acesupport/acemode/yaml_highlight_rules.js#L76-L90

**Comment by jmcphers** [comment]
Also coming up on support forums: https://community.rstudio.com/t/unexpected-token-issue-raw-string-in-r4-0-0/67947/2

**Comment by adamconroy** [comment]
Closed by #7447

**Comment by R-Hannibal** [comment]
@ajabadi, thank you for finding and reporting this improvement to the RStudio experience.

We have a fix in place in our latest Daily build of the upcoming 1.4 release of RStudio (available [HERE](https://dailies.rstudio.com/)). If you get the chance will you let us know if the request is resolved for you in that build?

**Comment by dfalty** [comment]
Verified fixed on version 1.4.874. @ajabadi let us know if you find other cases of this.
